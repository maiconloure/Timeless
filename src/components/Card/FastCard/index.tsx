import React from 'react';
import styled from 'styled-components';

interface FastCardProps {
  title: string;
  setTitle: any;
  subtitle: string;
  setSubtitle: any;
  time: string;
  setTime: any;
}

const FastCard = ({ title, setTitle, subtitle, setSubtitle, time, setTime }: FastCardProps) => {
  return (
    <Container>
      <Title type="text" value={title} onChange={(evt) => setTitle(evt.currentTarget.value)} />
      <SubTitle
        type="text"
        value={subtitle}
        onChange={(evt) => setSubtitle(evt.currentTarget.value)}
      />
      <Line />
      <Time type="text" value={time} onChange={(evt) => setTime(evt.currentTarget.value)} />
    </Container>
  );
};

export default FastCard;

const Container = styled.div`
  border: 2px solid #ff6347;
  background: #b22222;
  width: 200px;
  height: 100px;
  padding: 6px;
  margin: 5px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  border-radius: 2px;
  margin-left: 0;

  input {
    max-width: 180px;
    outline: none;
    border: none;
    background: none;
    color: #fff;
  }
`;

const Title = styled.input`
  font: 700 2.1rem Inter;
  color: #fff;
`;

const SubTitle = styled.input`
  font: 600 1.5rem Inter;
`;

const Line = styled.div`
  width: 180px;
  height: 2px;
  background: #fff;
  margin: 1px 0;
`;

const Time = styled.input`
  font: 600 2.1rem Inter;
  color: #fff;
`;
