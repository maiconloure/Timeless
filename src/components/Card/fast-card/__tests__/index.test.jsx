import React from 'react';

import renderer from 'react-test-renderer';
import FastCard from '../index';

const fastCard = {};

describe('Snapshot', () => {
  it('renderer', () => {
    const tree = renderer.create(<FastCard fastCard={{ fastCard }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
