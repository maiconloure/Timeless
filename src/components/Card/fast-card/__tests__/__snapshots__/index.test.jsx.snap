// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renderer 1`] = `
<div>
  <div
    className="sc-iJCRLp bwfUlN"
  >
    <div
      className="sc-giAqnE jBBlrY"
    >
      <h1
        className="sc-iklIKw bYHFxB"
        fontSize={16}
        fontStyle="italic"
        fontWeight={700}
      />
    </div>
    <h1
      className="sc-iklIKw fzaGyt"
      fontSize={12}
      fontStyle="italic"
      fontWeight={700}
    />
    <div
      className="sc-jJMGHv ilzVxZ"
    />
    <div
      className="sc-giAqnE drKKNL"
    >
      <div
        className="sc-bYwzba iwniLP"
      >
        <h1
          className="sc-iklIKw cYDQLW"
          fontSize={15}
          fontStyle="italic"
          fontWeight={700}
        />
      </div>
    </div>
  </div>
</div>
`;
